api/v1/submissions/PKPSubmissionHandler.php

```diff
@@ -477,45 +570,119 @@ public function add($slimRequest, $response, $args)
 
     /**
      * Edit a submission
-     *
-     * @param Request $slimRequest Slim request object
-     * @param Response $response object
-     * @param array $args arguments
-     *
-     * @return Response
      */
-    public function edit($slimRequest, $response, $args)
+    public function edit(SlimRequest $slimRequest, APIResponse $response, array $args): APIResponse
     {
         $request = $this->getRequest();
         $submission = $this->getAuthorizedContextObject(Application::ASSOC_TYPE_SUBMISSION);
 
-        if (!$submission) {
-            return $response->withStatus(404)->withJsonError('api.404.resourceNotFound');
+        $params = $this->convertStringsToSchema(PKPSchemaService::SCHEMA_SUBMISSION, $slimRequest->getParsedBody());
+
+        $readOnlyErrors = $this->getWriteDisabledErrors(PKPSchemaService::SCHEMA_SUBMISSION, $params);
+        if (!empty($readOnlyErrors)) {
+            return $response->withStatus(400)->withJson($readOnlyErrors);
         }
 
-        $params = $this->convertStringsToSchema(PKPSchemaService::SCHEMA_SUBMISSION, $slimRequest->getParsedBody());
         $params['id'] = $submission->getId();
         $params['contextId'] = $request->getContext()->getId();
 
-        // Don't allow a submission's context to be changed
-        if (isset($params['contextId']) && $params['contextId'] !== $submission->getData('contextId')) {
-            return $response->withStatus(403)->withJsonError('api.submissions.403.cantChangeContext');
```

Comment:
I removed the check because we now manually force the `contextId` to match the request, and `SubmissionRequiredPolicy` should only pass if the submission is in the current request.

(The `Repository::validate()` method only checks that the context exists, since it is meant to be able to run in the future without a request context, like from a CLI.)